{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TUBINGEN\n",
    "For Tübingen, extract all unique hgvs/vcf and annotate with VEP to get hgvs/vcf "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SACS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sacs = pd.read_csv(\"SACS/WP1/P1_SACS_raw.csv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Submitting institution\\n(select from dropdown)', 'local subject ID',\n",
       "       'local family ID', 'PROSPAX subject ID',\n",
       "       'NGS database ID (e.g. GENESIS, GPAP, etc)',\n",
       "       'Main phenotype \\n(select from dropdown)',\n",
       "       'Case solved? \\n(solved SACS / solved other gene / unsure / not solved)',\n",
       "       'Variant ID \\n(consecutive number)', 'Gene',\n",
       "       'Chromosome\\n(genome build hg19 / GRCh37)',\n",
       "       'Position\\n(use vcf format)', 'Reference allele', 'Alternative allele',\n",
       "       'Transcript\\n(use reference transcript ENST00000382298.3)',\n",
       "       'cDNA change (e.g. ', 'Protein position', 'Het / Hom',\n",
       "       'Compound heterozygous with Variant ID\\n(enter variant ID, 'N/A' or 'unknown')',\n",
       "       'PaxGene available?\\n(yes / no / unknown)',\n",
       "       'PBMC available?\\n(yes / no / unknown)',\n",
       "       'Fibroblasts available?\\n(yes / no / unknown)', 'Comment'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sacs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_var = df_sacs[['Variant ID \\n(consecutive number)', \n",
    "       'Chromosome\\n(genome build hg19 / GRCh37)',\n",
    "       'Position\\n(use vcf format)', 'Reference allele', 'Alternative allele',\n",
    "       'Transcript\\n(use reference transcript ENST00000382298.3)',\n",
    "       'cDNA change (e.g. ', 'Protein position']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID \\n(consecutive number)</th>\n",
       "      <th>Chromosome\\n(genome build hg19 / GRCh37)</th>\n",
       "      <th>Position\\n(use vcf format)</th>\n",
       "      <th>Reference allele</th>\n",
       "      <th>Alternative allele</th>\n",
       "      <th>Transcript\\n(use reference transcript ENST00000382298.3)</th>\n",
       "      <th>cDNA change (e.g.</th>\n",
       "      <th>Protein position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>#1</td>\n",
       "      <td>13</td>\n",
       "      <td>23914246.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>c.3769G&gt;T</td>\n",
       "      <td>p.Gly1275Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>#2</td>\n",
       "      <td>13</td>\n",
       "      <td>23909431.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>c.8584A&gt;T</td>\n",
       "      <td>p.Lys2862Ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>#3</td>\n",
       "      <td>13</td>\n",
       "      <td>23913292.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>c.4723C&gt;T</td>\n",
       "      <td>p.Arg1575Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>#4</td>\n",
       "      <td>13</td>\n",
       "      <td>23913292.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>c.4723C&gt;T</td>\n",
       "      <td>p.Arg1575Trp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>#5</td>\n",
       "      <td>13</td>\n",
       "      <td>23910851.0</td>\n",
       "      <td>GGT</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>c.7162_7163del</td>\n",
       "      <td>p.Thr2388Argfs*10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>108</td>\n",
       "      <td>#109</td>\n",
       "      <td>13</td>\n",
       "      <td>23932473.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>109</td>\n",
       "      <td>#110</td>\n",
       "      <td>13</td>\n",
       "      <td>23905042.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110</td>\n",
       "      <td>#111</td>\n",
       "      <td>13</td>\n",
       "      <td>23912269.0</td>\n",
       "      <td>CAT</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>111</td>\n",
       "      <td>#112</td>\n",
       "      <td>13</td>\n",
       "      <td>23905042.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112</td>\n",
       "      <td>#113</td>\n",
       "      <td>13</td>\n",
       "      <td>23912269.0</td>\n",
       "      <td>CAT</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000382298.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Variant ID \\n(consecutive number)  \\\n",
       "0                                  #1   \n",
       "1                                  #2   \n",
       "2                                  #3   \n",
       "3                                  #4   \n",
       "4                                  #5   \n",
       "..                                ...   \n",
       "108                              #109   \n",
       "109                              #110   \n",
       "110                              #111   \n",
       "111                              #112   \n",
       "112                              #113   \n",
       "\n",
       "     Chromosome\\n(genome build hg19 / GRCh37)  Position\\n(use vcf format)  \\\n",
       "0                                          13                  23914246.0   \n",
       "1                                          13                  23909431.0   \n",
       "2                                          13                  23913292.0   \n",
       "3                                          13                  23913292.0   \n",
       "4                                          13                  23910851.0   \n",
       "..                                        ...                         ...   \n",
       "108                                        13                  23932473.0   \n",
       "109                                        13                  23905042.0   \n",
       "110                                        13                  23912269.0   \n",
       "111                                        13                  23905042.0   \n",
       "112                                        13                  23912269.0   \n",
       "\n",
       "    Reference allele Alternative allele  \\\n",
       "0                  C                  A   \n",
       "1                  T                  A   \n",
       "2                  G                  A   \n",
       "3                  G                  A   \n",
       "4                GGT                  G   \n",
       "..               ...                ...   \n",
       "108                C                  T   \n",
       "109                G                  A   \n",
       "110              CAT                  C   \n",
       "111                G                  A   \n",
       "112              CAT                  C   \n",
       "\n",
       "    Transcript\\n(use reference transcript ENST00000382298.3)  \\\n",
       "0                                    ENST00000382298.3         \n",
       "1                                    ENST00000382298.3         \n",
       "2                                    ENST00000382298.3         \n",
       "3                                    ENST00000382298.3         \n",
       "4                                    ENST00000382298.3         \n",
       "..                                                 ...         \n",
       "108                                  ENST00000382298.3         \n",
       "109                                  ENST00000382298.3         \n",
       "110                                  ENST00000382298.3         \n",
       "111                                  ENST00000382298.3         \n",
       "112                                  ENST00000382298.3         \n",
       "\n",
       "    cDNA change (e.g.    Protein position  \n",
       "0            c.3769G>T       p.Gly1275Ter  \n",
       "1            c.8584A>T       p.Lys2862Ter  \n",
       "2            c.4723C>T       p.Arg1575Trp  \n",
       "3            c.4723C>T       p.Arg1575Trp  \n",
       "4       c.7162_7163del  p.Thr2388Argfs*10  \n",
       "..                 ...                ...  \n",
       "108                NaN                NaN  \n",
       "109                NaN                NaN  \n",
       "110                NaN                NaN  \n",
       "111                NaN                NaN  \n",
       "112                NaN                NaN  \n",
       "\n",
       "[113 rows x 8 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get empty hgvs lines\n",
    "# get empty vcf lines\n",
    "import numpy as np\n",
    "\n",
    "empty_hgvs = []\n",
    "empty_vcf = []\n",
    "for idx, row in d_var.iterrows():\n",
    "    if np.isnan(row['Position\\n(use vcf format)']):\n",
    "        empty_vcf.append(row['cDNA change (e.g. '])\n",
    "    elif np.isnan(row['cDNA change (e.g. ']):\n",
    "        vcf = \"{} {} . {} {} . . .\".format(row['Chromosome\\n(genome build hg19 / GRCh37)'],\n",
    "                                                                    int(row['Position\\n(use vcf format)']),\n",
    "                                                                    row['Reference allele'],\n",
    "                                                                    row['Alternative allele'])\n",
    "        empty_hgvs.append(vcf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SACS/WP1/empty_vcf.txt\", \"w\") as f:\n",
    "    for h in set(empty_vcf):\n",
    "        f.write(h + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SACS/WP1/empty_hgvs.txt\", \"w\") as f:\n",
    "    for h in set(empty_hgvs):\n",
    "        f.write(h + \"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cyvcf2 import VCF\n",
    "\n",
    "V_empty_vcf = VCF(\"SACS/WP1/P1_SACS_empty_VCF.vcf\")\n",
    "V_empty_hgvs = VCF(\"SACS/WP1/P1_sacs_empty_hgvs.vcf\")\n",
    "\n",
    "\n",
    "for v in V_empty_vcf:\n",
    "    hgvs = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0].split(\":\")[1]\n",
    "    for idx, row in df_sacs.iterrows():\n",
    "        if row['cDNA change (e.g. ']==hgvs:\n",
    "            df_sacs.at[idx, 'Position\\n(use vcf format)'] = v.POS\n",
    "            df_sacs.at[idx, 'Reference allele'] = v.REF\n",
    "            df_sacs.at[idx, 'Alternative allele'] = v.ALT[0]\n",
    "            \n",
    "            \n",
    "for v in V_empty_hgvs:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0].split(\":\")[1]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0]\n",
    "    \n",
    "    for idx, row in df_sacs.iterrows():\n",
    "        if row['Position\\n(use vcf format)']==v.POS and row['Reference allele']==v.REF and row['Alternative allele']==v.ALT[0]:\n",
    "            df_sacs.at[idx, 'cDNA change (e.g. '] = hgvs_c\n",
    "            df_sacs.at[idx, 'Protein position'] = hgvs_p\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sacs.to_csv(\"SACS/WP1/P1_SACS_1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Submitting institution\\n(select from dropdown)', 'local subject ID',\n",
       "       'local family ID', 'PROSPAX subject ID',\n",
       "       'NGS database ID (e.g. GENESIS, GPAP, etc)',\n",
       "       'Main phenotype \\n(select from dropdown)',\n",
       "       'Case solved? \\n(solved SACS / solved other gene / unsure / not solved)',\n",
       "       'Variant ID \\n(consecutive number)', 'Gene',\n",
       "       'Chromosome\\n(genome build hg19 / GRCh37)',\n",
       "       'Position\\n(use vcf format)', 'Reference allele', 'Alternative allele',\n",
       "       'Transcript\\n(use reference transcript ENST00000382298.3)',\n",
       "       'cDNA change (e.g. ', 'Protein position', 'Het / Hom',\n",
       "       'Compound heterozygous with Variant ID\\n(enter variant ID, 'N/A' or 'unknown')',\n",
       "       'PaxGene available?\\n(yes / no / unknown)',\n",
       "       'PBMC available?\\n(yes / no / unknown)',\n",
       "       'Fibroblasts available?\\n(yes / no / unknown)', 'Comment',\n",
       "       'Alternate allele'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df_sacs.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SPG7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Submitting institution\\n(select from dropdown)', 'local subject ID',\n",
       "       'local family ID', 'PROSPAX subject ID',\n",
       "       'NGS database ID (e.g. GENESIS, GPAP, etc)',\n",
       "       'Main phenotype \\n(select from dropdown)',\n",
       "       'Case solved? \\n(solved SPG7 / solved other gene / unsure / not solved)',\n",
       "       'Variant ID \\n(consecutive number)', 'Gene',\n",
       "       'Chromosome\\n(genome build hg19 / GRCh37)',\n",
       "       'Position\\n(use vcf format)', 'Reference allele', 'Alternative allele',\n",
       "       'Transcript\\n(use reference transcript ENST00000268704.2)',\n",
       "       'cDNA change (e.g. ', 'Protein position', 'Het / Hom',\n",
       "       'Compound heterozygous with Variant ID\\n(enter variant ID, 'N/A' or 'unknown')',\n",
       "       'PaxGene available?\\n(yes / no / unknown)',\n",
       "       'PBMC available?\\n(yes / no / unknown)',\n",
       "       'Fibroblasts available?\\n(yes / no / unknown)', 'Comment'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_spg7 = pd.read_csv(\"SPG7/WP1/P1_SPG7_raw.csv\", sep=\"\\t\")\n",
    "\n",
    "df_spg7.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get empty hgvs lines\n",
    "# get empty vcf lines\n",
    "import numpy as np\n",
    "\n",
    "empty_hgvs = []\n",
    "empty_vcf = []\n",
    "for idx, row in df_spg7.iterrows():\n",
    "    \n",
    "    if np.isnan(row['Position\\n(use vcf format)']):\n",
    "        empty_vcf.append(row['cDNA change (e.g. '])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = df_spg7['cDNA change (e.g. '].isnull()\n",
    "\n",
    "index_missing_hgvs = mask[mask==True].index\n",
    "\n",
    "for index in index_missing_hgvs:\n",
    "    pos = int(df_spg7.loc[index, 'Position\\n(use vcf format)'])\n",
    "    ref = df_spg7.loc[index, 'Reference allele']\n",
    "    alt = df_spg7.loc[index, 'Alternative allele']\n",
    "    \n",
    "    v = \"16 {} . {} {} . . .\\n\".format(pos, ref, alt)\n",
    "    empty_hgvs.append(v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SPG7/WP1/empty_vcf.txt\", \"w\") as f:\n",
    "    for h in set(empty_vcf):\n",
    "        f.write(h + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SPG7/WP1/empty_hgvs.txt\", \"w\") as f:\n",
    "    for h in set(empty_hgvs):\n",
    "        f.write(h.replace(\".0\",\"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cyvcf2 import VCF\n",
    "\n",
    "V_empty_vcf = VCF(\"SPG7/WP1/p1_spg7_empty_vcf.vcf\")\n",
    "V_empty_hgvs = VCF(\"SPG7/WP1/p1_spg7_empty_hgvs.vcf\")\n",
    "\n",
    "\n",
    "for v in V_empty_vcf:\n",
    "    hgvs = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0].split(\":\")[1]\n",
    "    for idx, row in df_spg7.iterrows():\n",
    "        if row['cDNA change (e.g. ']==hgvs:\n",
    "            df_spg7.at[idx, 'Position\\n(use vcf format)'] = v.POS\n",
    "            df_spg7.at[idx, 'Reference allele'] = v.REF\n",
    "            df_spg7.at[idx, 'Alternative allele'] = v.ALT[0]\n",
    "            \n",
    "            \n",
    "for v in V_empty_hgvs:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    \n",
    "    for idx, row in df_spg7.iterrows():\n",
    "        if row['Position\\n(use vcf format)']==v.POS and row['Reference allele']==v.REF and row['Alternative allele']==v.ALT[0]:\n",
    "            df_spg7.at[idx, 'cDNA change (e.g. '] = hgvs_c\n",
    "            df_spg7.at[idx, 'Protein position'] = hgvs_p\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_spg7.to_csv(\"SPG7/WP1/P1_SPG7_1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PISA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      NM_014363.6\n",
       "1      NM_014363.6\n",
       "2      NM_014363.6\n",
       "3      NM_014363.6\n",
       "4      NM_014363.6\n",
       "          ...     \n",
       "192    NM_014363.6\n",
       "193    NM_014363.6\n",
       "194    NM_014363.6\n",
       "195    NM_014363.6\n",
       "196    NM_014363.6\n",
       "Name: Transcript , Length: 197, dtype: object"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from cyvcf2 import VCF\n",
    "\n",
    "df_sacs_pisa = pd.read_csv(\"SACS/P4_pisa/SACS SPG7 Finale.xlsx - SACS.tsv\", sep=\"\\t\")\n",
    "df_spg7_pisa = pd.read_csv(\"SPG7/P4_pisa/SACS SPG7 Finale.xlsx - SPG7.tsv\", sep=\"\\t\")\n",
    "\n",
    "\n",
    "df_sacs_pisa[\"Diagnosis\"] = df_sacs_pisa[\"Diagnosis\"].fillna(\"unsure\")\n",
    "df_spg7_pisa[\"Diagnosis\"] = df_spg7_pisa[\"Diagnosis\"].fillna(\"unsure\")\n",
    "df_sacs_pisa[\"Transcript \"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference Allele</th>\n",
       "      <th>Sample Allele</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Transcript Variant</th>\n",
       "      <th>Protein Variant</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Codice</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>23907968</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NM_014363.6</td>\n",
       "      <td>c.10047A&gt;G</td>\n",
       "      <td>p.S3349S</td>\n",
       "      <td>Het</td>\n",
       "      <td>039C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>23915089</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NM_014363.6</td>\n",
       "      <td>c.2926C&gt;T</td>\n",
       "      <td>p.R976C</td>\n",
       "      <td>Het</td>\n",
       "      <td>039C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>23928078</td>\n",
       "      <td>C</td>\n",
       "      <td>/</td>\n",
       "      <td>NM_014363.6</td>\n",
       "      <td>c.2094-63delG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hom</td>\n",
       "      <td>119C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>23906661</td>\n",
       "      <td>CT</td>\n",
       "      <td>/</td>\n",
       "      <td>NM_014363.6</td>\n",
       "      <td>c.11353_11354delAG</td>\n",
       "      <td>p.R3785fs*24</td>\n",
       "      <td>Het</td>\n",
       "      <td>1652B</td>\n",
       "      <td>HSP</td>\n",
       "      <td>SPG7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>23912386</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NM_014363.6</td>\n",
       "      <td>c.5629C&gt;T</td>\n",
       "      <td>p.R1877*</td>\n",
       "      <td>Het</td>\n",
       "      <td>1652B</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome  Position Reference Allele Sample Allele  Transcript   \\\n",
       "0          13  23907968                T             C  NM_014363.6   \n",
       "1          13  23915089                G             A  NM_014363.6   \n",
       "2          13  23928078                C             /  NM_014363.6   \n",
       "3          13  23906661               CT             /  NM_014363.6   \n",
       "4          13  23912386                G             A  NM_014363.6   \n",
       "\n",
       "    Transcript Variant Protein Variant Genotype Codice Phenotype Diagnosis  \n",
       "0           c.10047A>G        p.S3349S      Het   039C       HSP    unsure  \n",
       "1            c.2926C>T         p.R976C      Het   039C       HSP    unsure  \n",
       "2        c.2094-63delG             NaN      Hom   119C       HSP    unsure  \n",
       "3   c.11353_11354delAG    p.R3785fs*24      Het  1652B       HSP      SPG7  \n",
       "4            c.5629C>T        p.R1877*      Het  1652B       HSP    unsure  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sacs_pisa.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference Allele</th>\n",
       "      <th>Sample Allele</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Transcript Variant</th>\n",
       "      <th>Protein Variant</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Codice</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>89574945.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NM_003119.4</td>\n",
       "      <td>c.120G&gt;A</td>\n",
       "      <td>p.G40G</td>\n",
       "      <td>Het</td>\n",
       "      <td>958C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>89619523.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_003119.4</td>\n",
       "      <td>c.1916C&gt;G</td>\n",
       "      <td>p.S639C</td>\n",
       "      <td>Het</td>\n",
       "      <td>958C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>89595897.0</td>\n",
       "      <td>TG</td>\n",
       "      <td>/</td>\n",
       "      <td>NM_003119.4</td>\n",
       "      <td>c.773_774delTG</td>\n",
       "      <td>p.V258fs*30</td>\n",
       "      <td>Het</td>\n",
       "      <td>701C</td>\n",
       "      <td>HSP</td>\n",
       "      <td>SPG7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>89592857.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NM_003119.4</td>\n",
       "      <td>c.739C&gt;T</td>\n",
       "      <td>p.R247*</td>\n",
       "      <td>Het</td>\n",
       "      <td>446B</td>\n",
       "      <td>HSP</td>\n",
       "      <td>SPG7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>89598369.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NM_003119.4</td>\n",
       "      <td>c.1045G&gt;A</td>\n",
       "      <td>p.G349S</td>\n",
       "      <td>Het</td>\n",
       "      <td>446B</td>\n",
       "      <td>HSP</td>\n",
       "      <td>unsure</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome    Position Reference Allele Sample Allele   Transcript  \\\n",
       "0          16  89574945.0                G             A  NM_003119.4   \n",
       "1          16  89619523.0                C             G  NM_003119.4   \n",
       "2          16  89595897.0               TG             /  NM_003119.4   \n",
       "3          16  89592857.0                C             T  NM_003119.4   \n",
       "4          16  89598369.0                G             A  NM_003119.4   \n",
       "\n",
       "  Transcript Variant Protein Variant Genotype Codice Phenotype Diagnosis  \n",
       "0           c.120G>A          p.G40G      Het   958C       HSP    unsure  \n",
       "1          c.1916C>G         p.S639C      Het   958C       HSP    unsure  \n",
       "2     c.773_774delTG     p.V258fs*30      Het   701C       HSP      SPG7  \n",
       "3           c.739C>T         p.R247*      Het   446B       HSP      SPG7  \n",
       "4          c.1045G>A         p.G349S      Het   446B       HSP    unsure  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_spg7_pisa.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NM_014363.6', 'NM_014363.9'], dtype=object)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sacs_pisa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NM_003119.4'], dtype=object)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_spg7_pisa[\"Transcript\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_hgvs_ref(row, tr_id):\n",
    "    for i in range(len(row[\"Transcript \"].split(\";\"))):\n",
    "        if row[\"Transcript \"].strip().startswith(tr_id):\n",
    "            variants = row[\"Transcript Variant\"].split(\";\")\n",
    "            return \"ENST00000382298.3:\" + variants[i].strip() + \"\\n\"\n",
    "    \n",
    "pisa_sacs_hgvs = []\n",
    "\n",
    "for idx, row in df_sacs_pisa.iterrows():\n",
    "    hgvs = get_hgvs_ref(row, \"NM_014363\")\n",
    "    patient = row[\"Codice\"]\n",
    "    diagnosis = row[\"Diagnosis\"]\n",
    "    phenotype = row[\"Phenotype\"]\n",
    "    genotype  = row[\"Genotype\"]\n",
    "    \n",
    "    \n",
    "    \n",
    "    pisa_sacs_hgvs.append(( patient, diagnosis, phenotype, genotype, hgvs))\n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "197"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pisa_sacs_hgvs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_hgvs_ref(row, tr_id):\n",
    "    for i in range(len(row[\"Transcript\"].split(\";\"))):\n",
    "        if row[\"Transcript\"].strip().startswith(tr_id):\n",
    "            variants = row[\"Transcript Variant\"].split(\";\")\n",
    "            return \"ENST00000268704.2:\" + variants[i].strip() + \"\\n\"\n",
    "    \n",
    "pisa_spg7_hgvs = []\n",
    "\n",
    "for idx, row in df_spg7_pisa.iterrows():\n",
    "    hgvs = get_hgvs_ref(row, \"NM_003119\")\n",
    "    patient = row[\"Codice\"]\n",
    "    diagnosis = row[\"Diagnosis\"]\n",
    "    phenotype = row[\"Phenotype\"]\n",
    "    genotype  = row[\"Genotype\"]\n",
    "    \n",
    "    \n",
    "    \n",
    "    pisa_spg7_hgvs.append(( patient, diagnosis, phenotype, genotype, hgvs))\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "298"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pisa_spg7_hgvs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "pisa_sacs_hgvs_c = list(set([entry[4] for entry in pisa_sacs_hgvs]))\n",
    "pisa_spg7_hgvs_c = list(set([entry[4] for entry in pisa_spg7_hgvs]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "129"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pisa_sacs_hgvs_c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "164"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pisa_spg7_hgvs_c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SACS/P4_pisa/sacs_hgvs_2_vep.txt\", \"w\") as f:\n",
    "    for h in pisa_sacs_hgvs_c:\n",
    "        f.write(h)\n",
    "        \n",
    "with open(\"SPG7/P4_pisa/spg7_hgvs_2_vep.txt\", \"w\") as f:\n",
    "    for h in pisa_spg7_hgvs_c:\n",
    "        f.write(h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cyvcf2 import VCF\n",
    "\n",
    "V_spg7_hgvs = VCF(\"SPG7/P4_pisa/P4_pisa_spg7_vep_v2.vcf\")\n",
    "V_sacs_hgvs = VCF(\"SACS/P4_pisa/P4_pisa_sacs_vep_v2.vcf\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('039C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.2926C>T\\n')"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pisa_sacs_hgvs[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "128\n",
      "156\n"
     ]
    }
   ],
   "source": [
    "count = 0 \n",
    "for line in V_sacs_hgvs:\n",
    "    count+=1\n",
    "    \n",
    "print(count)\n",
    "count2=0\n",
    "for line in V_spg7_hgvs:\n",
    "    count2+=1\n",
    "print(count2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('039C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.10047A>G\\n'),\n",
       " ('039C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.2926C>T\\n'),\n",
       " ('119C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.2094-63delG\\n'),\n",
       " ('1652B', 'SPG7', 'HSP', 'Het', 'ENST00000382298.3:c.11353_11354delAG\\n'),\n",
       " ('1652B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.5629C>T\\n'),\n",
       " ('1137B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.10982C>T\\n'),\n",
       " ('1137B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.2080G>A\\n'),\n",
       " ('543C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.7354T>C\\n'),\n",
       " ('543C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.4878C>T\\n'),\n",
       " ('543C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.260-54G>C\\n'),\n",
       " ('733B', 'SPG7', 'HSP', 'Het', 'ENST00000382298.3:c.6557_6559delGTA\\n'),\n",
       " ('733B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.9508C>T\\n'),\n",
       " ('730B',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Hom?',\n",
       "  'ENST00000382298.3:c.2186-622_2186-621insA\\n'),\n",
       " ('1864A', 'unsure', 'HSP', 'Hom?', 'ENST00000382298.3:c.2186-632delA\\n'),\n",
       " ('1864A', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.1878G>T\\n'),\n",
       " ('1864A', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.259+32C>T\\n'),\n",
       " ('1177B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.12738A>G\\n'),\n",
       " ('1177B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.2080G>A\\n'),\n",
       " ('1055B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.7231A>G\\n'),\n",
       " ('1055B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.2185+56T>A\\n'),\n",
       " ('546C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.9846A>G\\n'),\n",
       " ('546C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.259+28A>G\\n'),\n",
       " ('359A', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.1839G>A\\n'),\n",
       " ('359A', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.259+32C>T\\n'),\n",
       " ('38V', 'CYP2U1', 'HSP', 'Hom', 'ENST00000382298.3:c.*30_*31insA\\n'),\n",
       " ('1954V', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.6267G>A\\n'),\n",
       " ('1954V', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.4188C>T\\n'),\n",
       " (\"D'E\", 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " (\"D'E\", 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('RZT', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " ('RZT', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('2024X', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.*484T>C\\n'),\n",
       " ('709X', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.*484T>C\\n'),\n",
       " ('709X', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.3129A>G\\n'),\n",
       " ('1731X', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.*484T>C\\n'),\n",
       " ('1731X', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.13717A>C\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.6336A>C\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.6069C>T\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.5502G>C\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.4118C>T\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.4117G>C\\n'),\n",
       " ('1668C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.2185+56T>A\\n'),\n",
       " ('090C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.1839G>A\\n'),\n",
       " ('090C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.171+13C>T\\n'),\n",
       " ('081C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.2094-25G>A\\n'),\n",
       " ('366C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.3427C>A\\n'),\n",
       " ('366C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.2094-52A>G\\n'),\n",
       " ('361C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom?',\n",
       "  'ENST00000382298.3:c.259+74_259+76delTTT\\n'),\n",
       " ('361C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.259+73T>G\\n'),\n",
       " ('364B', 'unsure', 'CMT', 'Hom?', 'ENST00000382298.3:c.*30_*31insA\\n'),\n",
       " ('364B', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.12395T>C\\n'),\n",
       " ('1441B', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.9846A>G\\n'),\n",
       " ('1441B', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.259+28A>G\\n'),\n",
       " ('424B', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.11617G>A\\n'),\n",
       " ('424B', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.3907T>C\\n'),\n",
       " ('1579A', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.7504C>T\\n'),\n",
       " ('1579A', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.5151dupA\\n'),\n",
       " ('2436V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12218T>C\\n'),\n",
       " ('2436V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12213A>C\\n'),\n",
       " ('2436V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12212A>T\\n'),\n",
       " ('2436V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12211G>T\\n'),\n",
       " ('2436V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12206G>A\\n'),\n",
       " ('2431V', 'PMM2', 'HA', 'Het', 'ENST00000382298.3:c.7354T>A\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5473G>T\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5471T>A\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2216A>G/c.4466A>G\\n'),\n",
       " ('642X', 'RARS2', 'HA', 'Het', 'ENST00000382298.3:c.8648T>A\\n'),\n",
       " ('642X', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.6393T>C ?\\n'),\n",
       " ('371Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2988A>G\\n'),\n",
       " ('371Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2983G>T\\n'),\n",
       " ('470V', 'SPG7', 'HA', 'Het', 'ENST00000382298.3:c.2580A>G\\n'),\n",
       " ('2449V', 'CANVAS', 'HA', 'Hom', 'ENST00000382298.3:c.7548A>G\\n'),\n",
       " ('117A', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.7149C>T\\n'),\n",
       " ('117A', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.259+32C>T\\n'),\n",
       " ('338N', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.13717A>C\\n'),\n",
       " ('338N', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.171+3A>G\\n'),\n",
       " ('180Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.4717A>T\\n'),\n",
       " ('180Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2146C>T\\n'),\n",
       " ('180Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.1417A>G\\n'),\n",
       " ('180Z', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.605-99C>T\\n'),\n",
       " ('2431V', 'PMM2', 'HA', 'Het', 'ENST00000382298.3:c.12218T>C\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12213A>C\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12212A>T\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12211G>T\\n'),\n",
       " ('2431V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12206G>A\\n'),\n",
       " ('400Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12218T>C\\n'),\n",
       " ('400Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12213A>C\\n'),\n",
       " ('400Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12212A>T\\n'),\n",
       " ('400Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12211G>T\\n'),\n",
       " ('400Z', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12206G>A\\n'),\n",
       " ('642X', 'RARS2', 'HA', 'Het', 'ENST00000382298.3:c.12212A>T\\n'),\n",
       " ('642X', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12211G>T\\n'),\n",
       " ('642X', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.12206G>A\\n'),\n",
       " ('642X', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5473G>T\\n'),\n",
       " ('642X', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5471T>A\\n'),\n",
       " ('470V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5473G>T\\n'),\n",
       " ('470V', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5471T>A\\n'),\n",
       " ('745A', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5473G>T\\n'),\n",
       " ('745A', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.5471T>A\\n'),\n",
       " ('2576B ', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.13717A>C\\n'),\n",
       " ('2576B ', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.10982C>T\\n'),\n",
       " ('2477B', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.4430T>C\\n'),\n",
       " ('2477B', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2225G>C\\n'),\n",
       " ('1055B', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.7231A>G\\n'),\n",
       " ('1055B', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.2185+56T>A\\n'),\n",
       " ('2274t', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.1702G>A\\n'),\n",
       " ('2274t', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.20+32T>A\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.4757A>G\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.11048A>G\\n'),\n",
       " ('795C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.21-16541_21-16540delCA\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.21-16822A>G\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.21-16844C>T\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-501-2320C>A\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.-501-6902G>T\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-502+7246T>C\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-502+7148G>C\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-502+7087A>G\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.259+434G>A\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.8022T>C\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.457+1637C>T\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.457+774C>T\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.260-392A>C\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.260-402A>G\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.171+1057C>T\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.21-16429G>A\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.20+16642C>T\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.20+8256_20+8257insA\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.20+8232C>T\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.-501-2491T>G\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.-501-2797dupA\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.-502+7246T>C\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.-502+7148G>C\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.10611A>G\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.605-99C>T\\n'),\n",
       " ('362C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.21-16304_21-16303delGC\\n'),\n",
       " ('362C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.21-16541_21-16540delCA\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+15165T>C\\n'),\n",
       " ('362C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.20+13816_20+13820delGAGAT\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+11786C>T\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.11032C>G\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.457+1637C>T\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.260-392A>C\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.260-402A>G\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.260-433A>G\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.21-10591C>T\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.21-10606C>A\\n'),\n",
       " ('744C',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.21-16304_21-16303delGC\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.-501-1373C>A\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.9981T>C\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.171+6C>T\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.10338G>A\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.11032C>G\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.2185+3226T>A\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.2185+3225A>C\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.21-10591C>T\\n'),\n",
       " ('959C',\n",
       "  'unsure',\n",
       "  'CMT',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.21-16304_21-16303delGC\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+17180T>C\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+17128C>T\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+5837A>C\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+5833T>C\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.20+3385C>T\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-502+6895C>T\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000382298.3:c.-502+6885G>A\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.9981T>C\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.909A>G\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.10338G>A\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.*484T>C\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.2186-1870G>T\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.2185+2084A>G\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.21-10606C>A\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.21-16166A>G\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.21-16171T>C\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.20+15165T>C\\n'),\n",
       " ('1251C',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000382298.3:c.-502+6974_-502+6975delTT\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.-502+6895C>T\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.-502+6885G>A\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.-502+5950A>G\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000382298.3:c.-502+5810C>G\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.10042T>C\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.346-47G>A\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.10338G>A\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.696T>A\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.909A>G\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.605-99C>T\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.1656A>G\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.2186-1768C>T\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.457+774C>T\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.21-10194C>T\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Hom', 'ENST00000382298.3:c.21-17670G>A\\n'),\n",
       " ('2625B',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Het',\n",
       "  'ENST00000382298.3:c.20+8256_20+8257insA\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Het', 'ENST00000382298.3:c.20+8232C>T\\n')]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pisa_sacs_hgvs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sacs \n",
    "V_sacs_hgvs = VCF(\"SACS/P4_pisa/P4_pisa_sacs_vep_v2.vcf\")\n",
    "\n",
    "\n",
    "lines_f = []\n",
    "for v in V_sacs_hgvs:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0]\n",
    "    chrom = v.CHROM\n",
    "    pos = v.POS\n",
    "    ref = v.REF\n",
    "    alt = v.ALT[0]\n",
    "    \n",
    "    \n",
    "    for entry in pisa_sacs_hgvs:\n",
    "        if entry[4].strip()==hgvs_c:\n",
    "            sub = \"P4 - Pisa\"\n",
    "            local_id = entry[0]\n",
    "            f_id = \"F_\" + local_id\n",
    "            prospax = \"\"\n",
    "            ngs_db = \"Pisa Diagnostic Pipeline\"\n",
    "            main_phe = entry[2]\n",
    "            case_solved = entry[1]\n",
    "            v_id = \"\"\n",
    "            tr = \"ENST00000382298.3\"\n",
    "            hgvs_c_f = hgvs_c.split(\":\")[1]\n",
    "            genotype = entry[3]\n",
    "            \n",
    "            cmpd = \"\"\n",
    "            pax = \"\"\n",
    "            pbmc = \"\"\n",
    "            fibr = \"\"\n",
    "            comment = \"\"\n",
    "            \n",
    "            lines_f.append([sub, local_id, f_id, prospax, ngs_db, main_phe, case_solved,\n",
    "                          v_id, \"SACS\", chrom, pos, ref, alt, tr, hgvs_c_f, hgvs_p, genotype, cmpd,\n",
    "                          pbmc, pax, fibr, comment])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_row_manual(sub, local_id, f_id, ngs_db, main_phe, case_solved,\n",
    "                           gene, chrom, pos, ref, alt, tr, hgvsc, hgvsp, genotype, v_id=\"\", prospax=\"\",cmpd=\"\",\n",
    "                          pbmc=\"\", pax=\"\", fibr=\"\", comment=\"\"):\n",
    "    return [sub, local_id, f_id, prospax, ngs_db, main_phe, case_solved,\n",
    "                          v_id, gene, chrom, pos, ref, alt, tr, hgvsc, hgvsp, genotype, cmpd,\n",
    "                          pbmc, pax, fibr, comment]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "# manual variant\n",
    "missing = [h for h in pisa_sacs_hgvs if h[4].split(\":\")[1].strip() not in list(df_p4_sacs['cDNA change (e.g. '].unique())]\n",
    "for l in missing:\n",
    "    n_row = create_row_manual(\"P4 - Pisa\", l[0], \"F_\"+l[0], \"Pisa Diagnostic Pipeline\", l[2],\n",
    "                             l[1], \"SACS\", \"13\", \"\", \"\", \"\", \"ENST00000382298.3\", l[4],\n",
    "                             \"\", l[3])\n",
    "    lines_f.append(n_row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_p4_sacs = pd.DataFrame(lines_f, columns=list(df_sacs.columns)[:-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_p4_sacs.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(197, 22)"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_p4_sacs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_p4_sacs.to_csv(\"SACS/P4_pisa/P4_SACS_variant_collection.v1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ENST00000382298.3:c.259+74_259+76delTTT\\n',\n",
       " 'ENST00000382298.3:c.2186-632delA\\n',\n",
       " 'ENST00000382298.3:c.21-16304_21-16303delGC\\n',\n",
       " 'ENST00000382298.3:c.2186-622_2186-621insA\\n',\n",
       " 'ENST00000382298.3:c.2094-63delG\\n',\n",
       " 'ENST00000382298.3:c.2216A>G/c.4466A>G\\n',\n",
       " 'ENST00000382298.3:c.-501-2797dupA\\n',\n",
       " 'ENST00000382298.3:c.5151dupA\\n',\n",
       " 'ENST00000382298.3:c.6393T>C ?\\n',\n",
       " 'ENST00000382298.3:c.-502+6974_-502+6975delTT\\n',\n",
       " 'ENST00000382298.3:c.6557_6559delGTA\\n',\n",
       " 'ENST00000382298.3:c.21-16541_21-16540delCA\\n',\n",
       " 'ENST00000382298.3:c.20+13816_20+13820delGAGAT\\n',\n",
       " 'ENST00000382298.3:c.11353_11354delAG\\n']"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[h for h in pisa_sacs_hgvs_c if h.split(\":\")[1].strip() not in list(df_p4_sacs['cDNA change (e.g. '].unique())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "# spg7\n",
    "V_spg7_hgvs = VCF(\"SPG7/P4_pisa/P4_pisa_spg7_vep_v2.vcf\")\n",
    "\n",
    "lines_f_spg7 = []\n",
    "for v in V_spg7_hgvs:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    \n",
    "    chrom = v.CHROM\n",
    "    pos = v.POS\n",
    "    ref = v.REF\n",
    "    alt = v.ALT[0]\n",
    "    \n",
    "    for entry in pisa_spg7_hgvs:\n",
    "        if entry[4].strip()==hgvs_c:\n",
    "            sub = \"P4 - Pisa\"\n",
    "            local_id = entry[0]\n",
    "            f_id = \"F_\" + local_id\n",
    "            prospax = \"\"\n",
    "            ngs_db = \"Pisa Diagnostic Pipeline\"\n",
    "            main_phe = entry[2]\n",
    "            case_solved = entry[1]\n",
    "            v_id = \"\"\n",
    "            tr = \"ENST00000268704.2\"\n",
    "            hgvs_c_f = hgvs_c.split(\":\")[1]\n",
    "            genotype = entry[3]\n",
    "            \n",
    "            cmpd = \"\"\n",
    "            pax = \"\"\n",
    "            pbmc = \"\"\n",
    "            fibr = \"\"\n",
    "            comment = \"\"\n",
    "            \n",
    "            lines_f_spg7.append([sub, local_id, f_id, prospax, ngs_db, main_phe, case_solved,\n",
    "                          v_id, \"SPG7\", chrom, pos, ref, alt, tr, hgvs_c_f, hgvs_p, genotype, cmpd,\n",
    "                          pbmc, pax, fibr, comment])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(238, 22)"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_p4_spg7 = pd.DataFrame(lines_f_spg7, columns=list(df_sacs.columns)[:-1])\n",
    "df_p4_spg7.drop_duplicates(inplace=True)\n",
    "df_p4_spg7.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len([h for h in pisa_spg7_hgvs if h[4].split(\":\")[1].strip() not in list(df_p4_spg7['cDNA change (e.g. '].unique())])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [],
   "source": [
    "# manual variant\n",
    "missing = [h for h in pisa_spg7_hgvs if h[4].split(\":\")[1].strip() not in list(df_p4_spg7['cDNA change (e.g. '].unique())]\n",
    "for l in missing:\n",
    "    n_row = create_row_manual(\"P4 - Pisa\", l[0], \"F_\"+l[0], \"Pisa Diagnostic Pipeline\", l[2],\n",
    "                             l[1], \"SPG7\", \"16\", \"\", \"\", \"\", \"ENST00000268704.2\", l[4],\n",
    "                             \"\", l[3])\n",
    "    lines_f_spg7.append(n_row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(293, 22)"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_p4_spg7 = pd.DataFrame(lines_f_spg7, columns=list(df_spg7.columns))\n",
    "df_p4_spg7.drop_duplicates(inplace=True)\n",
    "df_p4_spg7.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_p4_spg7.to_csv(\"SPG7/P4_pisa/P4_SPG7_variant_collection.v1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "V_spg7_hgvs = VCF(\"SPG7/P4_pisa/P4_pisa_spg7_vep_v2.vcf\")\n",
    "ids = [v.ID for v in V_spg7_hgvs]\n",
    "miss = [h for h in pisa_spg7_hgvs if h[4].strip() not in ids]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('245B', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.-1G>A\\n'),\n",
       " ('052A', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1325-60A>G\\n'),\n",
       " ('1496A', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.*38G>T\\n'),\n",
       " ('2110T', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1325-20C>T\\n'),\n",
       " ('896X', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.51+751C>G\\n'),\n",
       " ('2781X', 'KCNA2', 'HA', 'Het', 'ENST00000268704.2:c.51+751C>G\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.51+537G>C\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.51+18C>G\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.51+18C>G\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.51+18C>G\\n')]"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('701C', 'SPG7', 'HSP', 'Het', 'ENST00000268704.2:c.773_774delTG\\n'),\n",
       " ('2418A',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Hom?',\n",
       "  'ENST00000268704.2:c.2181+155_2181+161delGTGTTCC\\n'),\n",
       " ('245B', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.-1G>A\\n'),\n",
       " ('2711A',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Hom?',\n",
       "  'ENST00000268704.2:c.184-19_184-18delGT\\n'),\n",
       " ('052A', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1325-60A>G\\n'),\n",
       " ('1496A', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.*38G>T\\n'),\n",
       " ('1443X',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Hom?',\n",
       "  'ENST00000268704.2:c.1780-17_1780-15delTCT\\n'),\n",
       " ('2110T', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1325-20C>T\\n'),\n",
       " ('1825X',\n",
       "  'unsure',\n",
       "  'HSP',\n",
       "  'Hom?',\n",
       "  'ENST00000268704.2:c.1324+136_1324+137insA\\n'),\n",
       " ('896X', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.51+751C>G\\n'),\n",
       " ('2781X', 'KCNA2', 'HA', 'Het', 'ENST00000268704.2:c.51+751C>G\\n'),\n",
       " ('2446V', 'SPG7', 'HA', 'Hom', 'ENST00000268704.2:c.73_80delCCAGGCCC\\n'),\n",
       " ('194Z',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.1780-17_1780-15delTCT\\n'),\n",
       " ('180Z',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.1325-160_1325-156delAGGTG\\n'),\n",
       " ('179Z',\n",
       "  'RNF216',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.988-131_988-114delTTTGAGGCTGCAGTGAGC\\n'),\n",
       " ('1175a ',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.1780-16_1780-14delCTT\\n'),\n",
       " ('729A', 'unsure', 'HA', 'Hom', 'ENST00000268704.2:c.988-74delA\\n'),\n",
       " ('716A',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.2181+155_2181+161delGTGTTCC\\n'),\n",
       " ('2272R', 'unsure', 'HA', 'Hom', 'ENST00000268704.2:c.184-19_184-17delGTT\\n'),\n",
       " ('1175a',\n",
       "  'unsure',\n",
       "  'HA',\n",
       "  'Hom',\n",
       "  'ENST00000268704.2:c.1780-16_1780-14delCTT\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Het', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Hom', 'ENST00000268704.2:c.377-4410delT\\n'),\n",
       " ('795C', 'unsure', 'CMT', 'Het', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.51+537G>C\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.376+726delT\\n'),\n",
       " ('694C', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000268704.2:c.377-3286delT\\n'),\n",
       " ('362C', 'unsure', 'CMT', 'Het', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Hom', 'ENST00000268704.2:c.376+4184dupG\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.1450-801dupT\\n'),\n",
       " ('744C', 'unsure', 'HA', 'Het', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.51+18C>G\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('873C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1450-801dupT\\n'),\n",
       " ('959C', 'unsure', 'CMT', 'Hom', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.51+18C>G\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1450-75delC\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('971C', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1450-801dupT\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('1251C', 'unsure', 'HA', 'Hom', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.992A>G\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.184-4delT\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('1594B', 'unsure', 'HSP', 'Hom', 'ENST00000268704.2:c.51+18C>G\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1324+148dupA\\n'),\n",
       " ('2625B', 'unsure', 'HSP', 'Het', 'ENST00000268704.2:c.1450-75delC\\n')]"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Nijmegen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sacs_nij = pd.read_csv(\"SACS/P3_nijmegen/SACS varianten MPEN BvdW 25-11.xlsx - Blad1.tsv\", sep=\"\\t\")\n",
    "df_spg7_nij = pd.read_csv(\"SPG7/P3_nijmegen/SPG7 varianten MPEN BvdW 25-11.xlsx - Blad1.tsv\", sep=\"\\t\")\n",
    "\n",
    "df_sacs_nij['Number of patients with this variant'] = df_sacs_nij['Number of patients with this variant'].fillna(' ')\n",
    "df_spg7_nij['Number of patients with this variant'] = df_spg7_nij['Number of patients with this variant'].fillna(' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Variant (genomic)', 'Variant ((non)coding)', 'Variant (rna/protein)',\n",
       "       'Classification', 'Phenotype', 'Number of patients with this variant',\n",
       "       'Other variant in this gene (if applicable)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sacs_nij.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Variant (genomic)', 'Variant ((non)coding)', 'Variant (rna/protein)',\n",
       "       'Classification', 'Phenotype', 'Number of patients with this variant',\n",
       "       'Other variant in this gene (if applicable)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_spg7_nij.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "nij_raw_rows_sacs = []\n",
    "for idx, row in df_sacs_nij.iterrows():\n",
    "    hgvsg = row['Variant (genomic)']\n",
    "    main_phenotype = row['Phenotype']\n",
    "    comment = \"nb=\"+row['Number of patients with this variant']+\";\"+\"comments=\"+row['Other variant in this gene (if applicable)']\n",
    "    \n",
    "    nij_raw_rows_sacs.append([hgvsg, main_phenotype, comment])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [],
   "source": [
    "nij_raw_rows_spg7 = []\n",
    "for idx, row in df_spg7_nij.iterrows():\n",
    "    hgvsg = row['Variant (genomic)']\n",
    "    main_phenotype = row['Phenotype']\n",
    "    comment = \"nb=\"+row['Number of patients with this variant']+\";\"+\"comments=\"+row['Other variant in this gene (if applicable)']\n",
    "    \n",
    "    nij_raw_rows_spg7.append([hgvsg, main_phenotype, comment])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"SACS/P3_nijmegen/SACS_2_vep.txt\", \"w\") as f:\n",
    "    for e in nij_raw_rows_sacs:\n",
    "        f.write(e[0] + \"\\n\")\n",
    "        \n",
    "with open(\"SPG7/P3_nijmegen/SPG7_2_vep.txt\", \"w\") as f:\n",
    "    for e in nij_raw_rows_spg7:\n",
    "        f.write(e[0] + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "V_nij_sacs = VCF(\"SACS/P3_nijmegen/P3_SACS_fromVep.vcf.vcf\")\n",
    "\n",
    "\n",
    "nij_lines_f_sacs = []\n",
    "for v in V_nij_sacs:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000382298.3\"][0]\n",
    "    \n",
    "    chrom = v.CHROM\n",
    "    pos = v.POS\n",
    "    ref = v.REF\n",
    "    alt = v.ALT[0]\n",
    "    \n",
    "    idx = v.ID\n",
    "    \n",
    "    for entry in nij_raw_rows_sacs:\n",
    "        if entry[0]==idx:\n",
    "            sub = \"P3 - Nijgemen\"\n",
    "            local_id = entry[0]\n",
    "            f_id = \"F_\" + local_id\n",
    "            prospax = \"\"\n",
    "            ngs_db = \"Nijgemen Diagnostic Pipeline\"\n",
    "            main_phe = entry[1]\n",
    "            case_solved = \"\"\n",
    "            v_id = \"\"\n",
    "            tr = \"ENST00000382298.3\"\n",
    "            hgvs_c_f = hgvs_c.split(\":\")[1]\n",
    "            genotype = \"\"\n",
    "            \n",
    "            cmpd = \"\"\n",
    "            pax = \"\"\n",
    "            pbmc = \"\"\n",
    "            fibr = \"\"\n",
    "            comment = entry[2]\n",
    "            \n",
    "            nij_lines_f_sacs.append([sub, local_id, f_id, prospax, ngs_db, main_phe, case_solved,\n",
    "                          v_id, \"SACS\", chrom, pos, ref, alt, tr, hgvs_c_f, hgvs_p, genotype, cmpd,\n",
    "                          pbmc, pax, fibr, comment])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "nij_sacs_df = pd.DataFrame(nij_lines_f_sacs, columns=df_p4_sacs.columns)\n",
    "\n",
    "nij_sacs_df.to_csv(\"SACS/P3_nijmegen/P3_Nijmegen_variant_collection.v1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "V_nij_spg7 = VCF(\"SPG7/P3_nijmegen/P3_SPG7_fromVep.vcf.vcf\")\n",
    "\n",
    "\n",
    "nij_lines_f_spg7 = []\n",
    "for v in V_nij_spg7:\n",
    "    hgvs_c = [tr.split(\"|\")[10] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    hgvs_p = [tr.split(\"|\")[11] for tr in v.INFO[\"CSQ\"].split(\",\") if tr.split(\"|\")[6] ==\"ENST00000268704.2\"][0]\n",
    "    \n",
    "    chrom = v.CHROM\n",
    "    pos = v.POS\n",
    "    ref = v.REF\n",
    "    alt = v.ALT[0]\n",
    "    \n",
    "    idx = v.ID\n",
    "    \n",
    "    for entry in nij_raw_rows_spg7:\n",
    "        if entry[0]==idx:\n",
    "            sub = \"P3 - Nijgemen\"\n",
    "            local_id = entry[0]\n",
    "            f_id = \"F_\" + local_id\n",
    "            prospax = \"\"\n",
    "            ngs_db = \"Nijgemen Diagnostic Pipeline\"\n",
    "            main_phe = entry[1]\n",
    "            case_solved = \"\"\n",
    "            v_id = \"\"\n",
    "            tr = \"ENST00000268704.2\"\n",
    "            hgvs_c_f = hgvs_c.split(\":\")[1]\n",
    "            genotype = \"\"\n",
    "            \n",
    "            cmpd = \"\"\n",
    "            pax = \"\"\n",
    "            pbmc = \"\"\n",
    "            fibr = \"\"\n",
    "            comment = entry[2]\n",
    "            \n",
    "            nij_lines_f_spg7.append([sub, local_id, f_id, prospax, ngs_db, main_phe, case_solved,\n",
    "                          v_id, \"SPG7\", chrom, pos, ref, alt, tr, hgvs_c_f, hgvs_p, genotype, cmpd,\n",
    "                          pbmc, pax, fibr, comment])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nij_raw_rows_spg7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nij_lines_f_spg7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [],
   "source": [
    "nij_spg7_df = pd.DataFrame(nij_lines_f_spg7, columns=df_p4_spg7.columns)\n",
    "\n",
    "nij_spg7_df.to_csv(\"SPG7/P3_nijmegen/P3_Nijmegen_variant_collection.v1.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant (genomic)</th>\n",
       "      <th>Variant ((non)coding)</th>\n",
       "      <th>Variant (rna/protein)</th>\n",
       "      <th>Classification</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Number of patients with this variant</th>\n",
       "      <th>Other variant in this gene (if applicable)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Chr16(GRCh37):g.89574828G&gt;A</td>\n",
       "      <td>c.3G&gt;A</td>\n",
       "      <td>p.(Met1?)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Chr16(GRCh37):g.89574859C&gt;T</td>\n",
       "      <td>c.34C&gt;T</td>\n",
       "      <td>p.(Arg12Cys)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Chr16(GRCh37):g.89574971C&gt;T</td>\n",
       "      <td>c.146C&gt;T</td>\n",
       "      <td>p.(Pro49Leu)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd (not highly conserved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Chr16(GRCh37):g.89579404_89579405insTA</td>\n",
       "      <td>c.335_336insTA</td>\n",
       "      <td>p.(Glu112fs)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>spinocerebellar ataxia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (with c.1324G&gt;C)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Chr16(GRCh37):g.89590648G&gt;T</td>\n",
       "      <td>c.611G&gt;T</td>\n",
       "      <td>p.(Gly204Val)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Chr16(GRCh37):g.89592812del</td>\n",
       "      <td>c.694del</td>\n",
       "      <td>p.(Glu232fs)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>spastic ataxia, cerebral visual disturbances</td>\n",
       "      <td></td>\n",
       "      <td>comp het. (with c.1529C&gt;T)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Chr16(GRCh37):g.89595987dup</td>\n",
       "      <td>c.861dup</td>\n",
       "      <td>p.((Asn288*)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia / progressive bipyramidal s...</td>\n",
       "      <td>4x</td>\n",
       "      <td>no 2nd / comp het. (c.1045G&gt;A) / comp het. del...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Chr16(GRCh37):g.89595989dup</td>\n",
       "      <td>c.861+2dup</td>\n",
       "      <td>r.(spl?)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>3x</td>\n",
       "      <td>2x comp het. (c.1045G&gt;A) / No 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Chr16(GRCh37):g.89598270_89598370del</td>\n",
       "      <td>c.988-42_1046del</td>\n",
       "      <td>r.(spl?)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1454_1462del)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Chr16(GRCh37):g.89598369G&gt;A</td>\n",
       "      <td>c.1045G&gt;A</td>\n",
       "      <td>p.(Gly349Ser)</td>\n",
       "      <td>class 4</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>found many times (&gt;20)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Chr16(GRCh37):g.89598372C&gt;A</td>\n",
       "      <td>c.1048C&gt;A</td>\n",
       "      <td>p.(Pro350Thr)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Chr16(GRCh37):g.89598372C&gt;T</td>\n",
       "      <td>c.1048C&gt;T</td>\n",
       "      <td>p.(Pro350Ser)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Chr16(GRCh37):g.89598377dup</td>\n",
       "      <td>c.1053dup</td>\n",
       "      <td>p.(Gly352fs)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia / cerebellar Ataxia</td>\n",
       "      <td>2x</td>\n",
       "      <td>comp het. (c.1749G&gt;C) /  comp het. (c.2228T&gt;C)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Chr16(GRCh37):g.89598912C&gt;T</td>\n",
       "      <td>c.1192C&gt;T</td>\n",
       "      <td>p.(Arg398*)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1529C&gt;T)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Chr16(GRCh37):g.89599025G&gt;T</td>\n",
       "      <td>c.1305G&gt;T</td>\n",
       "      <td>p.(Gln435His)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>3x</td>\n",
       "      <td>comp het. (c.2275G&gt;A)/ comp het. (c.Ala749Thr)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Chr16(GRCh37):g.89599044G&gt;C</td>\n",
       "      <td>c.1324G&gt;C</td>\n",
       "      <td>p.(Gly442Arg)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>spinocerebellar ataxia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het (c.335_336insTA)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Chr16(GRCh37):g.89613069A&gt;G</td>\n",
       "      <td>c.1453A&gt;G</td>\n",
       "      <td>p.(Arg485Gly)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Chr16(GRCh37):g.89613070_89613078del</td>\n",
       "      <td>c.1454_1462del</td>\n",
       "      <td>p.(Arg485_Glu487del)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>&gt;20</td>\n",
       "      <td>found many times (&gt;20)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Chr16(GRCh37):g.89613145C&gt;T</td>\n",
       "      <td>c.1529C&gt;T</td>\n",
       "      <td>p.(Ala510Val)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>&gt;20</td>\n",
       "      <td>found many times (&gt;20)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Chr16(GRCh37):g.89613159G&gt;C</td>\n",
       "      <td>c.1543G&gt;C</td>\n",
       "      <td>p.(Gly515Arg)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Chr16(GRCh37):g.89614409_89614410del</td>\n",
       "      <td>c.1553-2_1553-1del</td>\n",
       "      <td>r.(spl?)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Ataxia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Chr16(GRCh37):g.89614459G&gt;A</td>\n",
       "      <td>c.1601G&gt;A</td>\n",
       "      <td>p.(Arg534Gln)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Chr16(GRCh37):g.89616940C&gt;T</td>\n",
       "      <td>c.1702C&gt;T</td>\n",
       "      <td>p.(Gln568*)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1045G&gt;A)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Chr16(GRCh37):g.89616987G&gt;C</td>\n",
       "      <td>c.1749G&gt;C</td>\n",
       "      <td>p.(Trp583Cys)</td>\n",
       "      <td>class 4</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1053dup)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Chr16(GRCh37):g.89616994G&gt;T</td>\n",
       "      <td>c.1756G&gt;T</td>\n",
       "      <td>p.(Glu586*)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Ataxia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1529C&gt;T)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Chr16(GRCh37):g.89619403G&gt;T</td>\n",
       "      <td>c.1796G&gt;T</td>\n",
       "      <td>p.(Arg599Leu)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Chr16(GRCh37):g.89619501G&gt;A</td>\n",
       "      <td>c.1894G&gt;A</td>\n",
       "      <td>p.(Gly632Arg)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>2x</td>\n",
       "      <td>both comp het. (c.1529C&gt;T)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Chr16(GRCh37):g.89620213G&gt;T</td>\n",
       "      <td>c.1948G&gt;T</td>\n",
       "      <td>p.(Asp650Tyr)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Chr16(GRCh37):g.89620348C&gt;G</td>\n",
       "      <td>c.2083C&gt;G</td>\n",
       "      <td>p.(Leu695Val)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Chr16(GRCh37):g.89620355A&gt;C</td>\n",
       "      <td>c.2090A&gt;C</td>\n",
       "      <td>p.(Gln697Pro)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1045G&gt;A)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Chr16(GRCh37):g.89620367A&gt;C</td>\n",
       "      <td>c.2102A&gt;C</td>\n",
       "      <td>p.(His701Pro)</td>\n",
       "      <td>class 4</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>3x</td>\n",
       "      <td>homozygous / comp het. (c.1454_1462del) / no 2nd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Chr16(GRCh37):g.89620905_89620921del</td>\n",
       "      <td>c.2115_2131del</td>\n",
       "      <td>p.(Leu706fs)</td>\n",
       "      <td>class 5</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>4x</td>\n",
       "      <td>comp het. (1045G&gt;A) / comp het. (c.1454_1462de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Chr16(GRCh37):g.89623341T&gt;C</td>\n",
       "      <td>c.2228T&gt;C</td>\n",
       "      <td>p.(Ile743Thr)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.861dup)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>Chr16(GRCh37):g.89623353T&gt;C</td>\n",
       "      <td>c.2240T&gt;C</td>\n",
       "      <td>p.(Ile747Thr)</td>\n",
       "      <td>class 3</td>\n",
       "      <td>Spastic paraplegia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>comp het. (c.1454_1462del)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Variant (genomic) Variant ((non)coding)  \\\n",
       "0              Chr16(GRCh37):g.89574828G>A                c.3G>A   \n",
       "1              Chr16(GRCh37):g.89574859C>T               c.34C>T   \n",
       "2              Chr16(GRCh37):g.89574971C>T              c.146C>T   \n",
       "3   Chr16(GRCh37):g.89579404_89579405insTA        c.335_336insTA   \n",
       "4              Chr16(GRCh37):g.89590648G>T              c.611G>T   \n",
       "5              Chr16(GRCh37):g.89592812del              c.694del   \n",
       "6              Chr16(GRCh37):g.89595987dup              c.861dup   \n",
       "7              Chr16(GRCh37):g.89595989dup            c.861+2dup   \n",
       "8     Chr16(GRCh37):g.89598270_89598370del      c.988-42_1046del   \n",
       "9              Chr16(GRCh37):g.89598369G>A             c.1045G>A   \n",
       "10             Chr16(GRCh37):g.89598372C>A             c.1048C>A   \n",
       "11             Chr16(GRCh37):g.89598372C>T             c.1048C>T   \n",
       "12             Chr16(GRCh37):g.89598377dup             c.1053dup   \n",
       "13             Chr16(GRCh37):g.89598912C>T             c.1192C>T   \n",
       "14             Chr16(GRCh37):g.89599025G>T             c.1305G>T   \n",
       "15             Chr16(GRCh37):g.89599044G>C             c.1324G>C   \n",
       "16             Chr16(GRCh37):g.89613069A>G             c.1453A>G   \n",
       "17    Chr16(GRCh37):g.89613070_89613078del        c.1454_1462del   \n",
       "18             Chr16(GRCh37):g.89613145C>T             c.1529C>T   \n",
       "19             Chr16(GRCh37):g.89613159G>C             c.1543G>C   \n",
       "20    Chr16(GRCh37):g.89614409_89614410del    c.1553-2_1553-1del   \n",
       "21             Chr16(GRCh37):g.89614459G>A             c.1601G>A   \n",
       "22             Chr16(GRCh37):g.89616940C>T             c.1702C>T   \n",
       "23             Chr16(GRCh37):g.89616987G>C             c.1749G>C   \n",
       "24             Chr16(GRCh37):g.89616994G>T             c.1756G>T   \n",
       "25             Chr16(GRCh37):g.89619403G>T             c.1796G>T   \n",
       "26             Chr16(GRCh37):g.89619501G>A             c.1894G>A   \n",
       "27             Chr16(GRCh37):g.89620213G>T             c.1948G>T   \n",
       "28             Chr16(GRCh37):g.89620348C>G             c.2083C>G   \n",
       "29             Chr16(GRCh37):g.89620355A>C             c.2090A>C   \n",
       "30             Chr16(GRCh37):g.89620367A>C             c.2102A>C   \n",
       "31    Chr16(GRCh37):g.89620905_89620921del        c.2115_2131del   \n",
       "32             Chr16(GRCh37):g.89623341T>C             c.2228T>C   \n",
       "33             Chr16(GRCh37):g.89623353T>C             c.2240T>C   \n",
       "\n",
       "   Variant (rna/protein) Classification  \\\n",
       "0              p.(Met1?)        class 5   \n",
       "1           p.(Arg12Cys)        class 3   \n",
       "2           p.(Pro49Leu)        class 3   \n",
       "3           p.(Glu112fs)        class 5   \n",
       "4          p.(Gly204Val)        class 3   \n",
       "5           p.(Glu232fs)        class 5   \n",
       "6           p.((Asn288*)        class 5   \n",
       "7               r.(spl?)        class 5   \n",
       "8               r.(spl?)        class 5   \n",
       "9          p.(Gly349Ser)        class 4   \n",
       "10         p.(Pro350Thr)        class 3   \n",
       "11         p.(Pro350Ser)        class 3   \n",
       "12          p.(Gly352fs)        class 5   \n",
       "13           p.(Arg398*)        class 5   \n",
       "14         p.(Gln435His)        class 3   \n",
       "15         p.(Gly442Arg)        class 3   \n",
       "16         p.(Arg485Gly)        class 3   \n",
       "17  p.(Arg485_Glu487del)        class 5   \n",
       "18         p.(Ala510Val)        class 5   \n",
       "19         p.(Gly515Arg)        class 3   \n",
       "20              r.(spl?)        class 5   \n",
       "21         p.(Arg534Gln)        class 3   \n",
       "22           p.(Gln568*)        class 5   \n",
       "23         p.(Trp583Cys)        class 4   \n",
       "24           p.(Glu586*)        class 5   \n",
       "25         p.(Arg599Leu)        class 3   \n",
       "26         p.(Gly632Arg)        class 3   \n",
       "27         p.(Asp650Tyr)        class 3   \n",
       "28         p.(Leu695Val)        class 3   \n",
       "29         p.(Gln697Pro)        class 3   \n",
       "30         p.(His701Pro)        class 4   \n",
       "31          p.(Leu706fs)        class 5   \n",
       "32         p.(Ile743Thr)        class 3   \n",
       "33         p.(Ile747Thr)        class 3   \n",
       "\n",
       "                                            Phenotype  \\\n",
       "0                                  Spastic paraplegia   \n",
       "1                                  Spastic paraplegia   \n",
       "2                                  Spastic paraplegia   \n",
       "3                              spinocerebellar ataxia   \n",
       "4                                 spastic paraplegia    \n",
       "5        spastic ataxia, cerebral visual disturbances   \n",
       "6   Spastic paraplegia / progressive bipyramidal s...   \n",
       "7                                  Spastic paraplegia   \n",
       "8                                  Spastic paraplegia   \n",
       "9                                  Spastic paraplegia   \n",
       "10                                 Spastic paraplegia   \n",
       "11                                 Spastic paraplegia   \n",
       "12             Spastic paraplegia / cerebellar Ataxia   \n",
       "13                                 Spastic paraplegia   \n",
       "14                                 Spastic paraplegia   \n",
       "15                             spinocerebellar ataxia   \n",
       "16                                 Spastic paraplegia   \n",
       "17                                 Spastic paraplegia   \n",
       "18                                 Spastic paraplegia   \n",
       "19                                 Spastic paraplegia   \n",
       "20                                             Ataxia   \n",
       "21                                 Spastic paraplegia   \n",
       "22                                 Spastic paraplegia   \n",
       "23                                 Spastic paraplegia   \n",
       "24                                             Ataxia   \n",
       "25                                 Spastic paraplegia   \n",
       "26                                 Spastic paraplegia   \n",
       "27                                 Spastic paraplegia   \n",
       "28                                 Spastic paraplegia   \n",
       "29                                 Spastic paraplegia   \n",
       "30                                 Spastic paraplegia   \n",
       "31                                 Spastic paraplegia   \n",
       "32                                 Spastic paraplegia   \n",
       "33                                 Spastic paraplegia   \n",
       "\n",
       "   Number of patients with this variant  \\\n",
       "0                                   NaN   \n",
       "1                                   NaN   \n",
       "2                                   NaN   \n",
       "3                                   NaN   \n",
       "4                                   NaN   \n",
       "5                                         \n",
       "6                                    4x   \n",
       "7                                    3x   \n",
       "8                                   NaN   \n",
       "9                                   NaN   \n",
       "10                                  NaN   \n",
       "11                                  NaN   \n",
       "12                                   2x   \n",
       "13                                  NaN   \n",
       "14                                   3x   \n",
       "15                                  NaN   \n",
       "16                                  NaN   \n",
       "17                                  >20   \n",
       "18                                  >20   \n",
       "19                                  NaN   \n",
       "20                                  NaN   \n",
       "21                                  NaN   \n",
       "22                                  NaN   \n",
       "23                                  NaN   \n",
       "24                                  NaN   \n",
       "25                                  NaN   \n",
       "26                                   2x   \n",
       "27                                  NaN   \n",
       "28                                  NaN   \n",
       "29                                  NaN   \n",
       "30                                   3x   \n",
       "31                                   4x   \n",
       "32                                  NaN   \n",
       "33                                  NaN   \n",
       "\n",
       "           Other variant in this gene (if applicable)  \n",
       "0                                          homozygous  \n",
       "1                                              no 2nd  \n",
       "2                       no 2nd (not highly conserved)  \n",
       "3                          comp het. (with c.1324G>C)  \n",
       "4                                              no 2nd  \n",
       "5                          comp het. (with c.1529C>T)  \n",
       "6   no 2nd / comp het. (c.1045G>A) / comp het. del...  \n",
       "7                 2x comp het. (c.1045G>A) / No 2nd    \n",
       "8                          comp het. (c.1454_1462del)  \n",
       "9                              found many times (>20)  \n",
       "10                                             no 2nd  \n",
       "11                                             no 2nd  \n",
       "12     comp het. (c.1749G>C) /  comp het. (c.2228T>C)  \n",
       "13                              comp het. (c.1529C>T)  \n",
       "14  comp het. (c.2275G>A)/ comp het. (c.Ala749Thr)...  \n",
       "15                          comp het (c.335_336insTA)  \n",
       "16                                         homozygous  \n",
       "17                             found many times (>20)  \n",
       "18                             found many times (>20)  \n",
       "19                                             no 2nd  \n",
       "20                                             no 2nd  \n",
       "21                                             no 2nd  \n",
       "22                              comp het. (c.1045G>A)  \n",
       "23                              comp het. (c.1053dup)  \n",
       "24                              comp het. (c.1529C>T)  \n",
       "25                                             no 2nd  \n",
       "26                         both comp het. (c.1529C>T)  \n",
       "27                                         homozygous  \n",
       "28                                             no 2nd  \n",
       "29                              comp het. (c.1045G>A)  \n",
       "30   homozygous / comp het. (c.1454_1462del) / no 2nd  \n",
       "31  comp het. (1045G>A) / comp het. (c.1454_1462de...  \n",
       "32                               comp het. (c.861dup)  \n",
       "33                         comp het. (c.1454_1462del)  "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_spg7_nij"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests, sys, json\n",
    "from pprint import pprint\n",
    "\n",
    "def fetch_endpoint(server, request, content_type):\n",
    "\n",
    "    r = requests.get(server+request, headers={ \"Accept\" : content_type})\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "\n",
    "def fetch_endpoint_POST(server, request, data, content_type):\n",
    "\n",
    "    r = requests.post(server+request,\n",
    "                      headers={ \"Accept\" : content_type},\n",
    "                      data=data )\n",
    "\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    if content_type == 'application/json':\n",
    "        return r.json()\n",
    "    else:\n",
    "        return r.text\n",
    "\n",
    "# define the server, extension and content type\n",
    "server = \"http://rest.ensembl.org/\"\n",
    "con = \"application/json\"\n",
    "vep_ext = \"vep/homo_sapiens/hgvs/vcf=-\"\n",
    "\n",
    "# create the list of HGVS annotations\n",
    "hgvs = [\"SACS:c.2094-63delG\",\"SACS:c.11353_11354delAG\",\"SACS:c.6557_6559delGTA\",\n",
    "        \"SACS:c.2186-622_2186-621insA\",\"SACS:c.2186-632delA\",\"SACS:c.259+74_259+76delTTT\",\n",
    "        \"SACS:c.5151dupA\",\"SACS:c.2216A>G/c.4466A>G\",\"SACS:c.6393T>C\",\"SACS:c.21-16541_21-16540delCA\",\n",
    "        \"SACS:c.-501-2797dupA\",\"SACS:c.21-16304_21-16303delGC\",\"SACS:c.21-16541_21-16540delCA\",\n",
    "        \"SACS:c.20+13816_20+13820delGAGAT\",\"SACS:c.21-16304_21-16303delGC\",\"SACS:c.21-16304_21-16303delGC\",\n",
    "        \"SACS:c.-502+6974_-502+6975delTT\"\n",
    "]\n",
    "\n",
    "# convert the list into json format\n",
    "hgvs_json = json.dumps({ \"hgvs_notations\" : hgvs })\n",
    "\n",
    "# run the query\n",
    "post_vep = fetch_endpoint_POST(server, vep_ext, hgvs_json, con)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'SACS:c.2094-63delG',\n",
       " 'end': 23353939,\n",
       " 'input': 'SACS:c.2094-63delG',\n",
       " 'most_severe_consequence': 'intron_variant',\n",
       " 'transcript_consequences': [{'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000382292',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'protein_coding',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'transcript_id': 'ENST00000402364',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'protein_coding'},\n",
       "  {'hgnc_id': 'HGNC:10519',\n",
       "   'biotype': 'protein_coding',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'transcript_id': 'ENST00000423156',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'impact': 'MODIFIER',\n",
       "   'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS'},\n",
       "  {'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'biotype': 'protein_coding',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000455470',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'impact': 'MODIFIER',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000476776',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'processed_transcript',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'hgnc_id': 'HGNC:10519',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000682775',\n",
       "   'impact': 'MODIFIER',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'biotype': 'protein_coding',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835'},\n",
       "  {'hgnc_id': 'HGNC:10519',\n",
       "   'biotype': 'protein_coding',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'transcript_id': 'ENST00000682944',\n",
       "   'impact': 'MODIFIER',\n",
       "   'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS'},\n",
       "  {'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'distance': 29,\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['downstream_gene_variant'],\n",
       "   'transcript_id': 'ENST00000683154',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'retained_intron',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'consequence_terms': ['intron_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'impact': 'MODIFIER',\n",
       "   'transcript_id': 'ENST00000683210',\n",
       "   'biotype': 'protein_coding',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1},\n",
       "  {'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'transcript_id': 'ENST00000683270',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'protein_coding'},\n",
       "  {'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'biotype': 'protein_coding',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000683367',\n",
       "   'impact': 'MODIFIER',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'biotype': 'protein_coding',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000683489',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'impact': 'MODIFIER',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'biotype': 'protein_coding',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000683680',\n",
       "   'impact': 'MODIFIER'},\n",
       "  {'hgnc_id': 'HGNC:10519',\n",
       "   'transcript_id': 'ENST00000684163',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'protein_coding',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835'},\n",
       "  {'gene_id': 'ENSG00000151835',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'biotype': 'retained_intron',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'],\n",
       "   'transcript_id': 'ENST00000684196',\n",
       "   'impact': 'MODIFIER',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1,\n",
       "   'transcript_id': 'ENST00000684325',\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'consequence_terms': ['intron_variant', 'NMD_transcript_variant'],\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'nonsense_mediated_decay',\n",
       "   'hgnc_id': 'HGNC:10519'},\n",
       "  {'strand': -1,\n",
       "   'variant_allele': '-',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000684385',\n",
       "   'impact': 'MODIFIER',\n",
       "   'biotype': 'protein_coding'},\n",
       "  {'gene_symbol_source': 'HGNC',\n",
       "   'transcript_id': 'ENST00000684497',\n",
       "   'impact': 'MODIFIER',\n",
       "   'consequence_terms': ['intron_variant'],\n",
       "   'biotype': 'protein_coding',\n",
       "   'hgnc_id': 'HGNC:10519',\n",
       "   'gene_symbol': 'SACS',\n",
       "   'gene_id': 'ENSG00000151835',\n",
       "   'variant_allele': '-',\n",
       "   'strand': -1}],\n",
       " 'assembly_name': 'GRCh38',\n",
       " 'seq_region_name': '13',\n",
       " 'allele_string': 'G/-',\n",
       " 'strand': -1,\n",
       " 'start': 23353939,\n",
       " 'colocated_variants': [{'strand': 1,\n",
       "   'frequencies': {'-': {'sas': 0,\n",
       "     'afr': 0.0333,\n",
       "     'amr': 0.0043,\n",
       "     'eur': 0,\n",
       "     'eas': 0}},\n",
       "   'minor_allele_freq': 0.0094,\n",
       "   'start': 23353939,\n",
       "   'minor_allele': '-',\n",
       "   'allele_string': 'C/-',\n",
       "   'seq_region_name': '13',\n",
       "   'end': 23353939,\n",
       "   'id': 'rs113268445'}]}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "post_vep[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# move through the results\n",
    "for variant in post_vep:\n",
    "    \n",
    "    # get the data\n",
    "    input = variant['input']\n",
    "    colocated_list = []\n",
    "    for colocated in variant['colocated_variants']:\n",
    "        colocated_list.append(colocated['id'])  \n",
    "    print (input + \": \" + (', '.join(colocated_list)))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
